connectors/source.py

```diff
@@ -384,7 +384,7 @@ class BaseDataSource:
     advanced_rules_enabled = False
     dls_enabled = False
     incremental_sync_enabled = False
-    native_connector_api_keys_enabled = True
+    native_connector_api_keys_enabled = False
```

Comment:
What happens here?

Reply:
This is old setting to enable `native connector secrets` for native connectors. Since we no longer rely on secrets for elastic managed connectors I just set this flag to false without thinking worrying about proper cleanup, I've filed followup ticket:
https://github.com/elastic/connectors/issues/2990

Reply:
So it'll break connectors running in native mode in traditional setup in 9.x, but we don't care since it's something nobody's supposed to do, right?

Reply:
The native connectors would be broken anyway since only service account that manages the ent-search node has  access to read those https://github.com/elastic/elasticsearch/blob/8c20ac5884158b88fdd598e422db632e1734aabb/x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/service/ElasticServiceAccounts.java#L48 

After migration to 9.0 those secrets are useless because noone has permissions to read them